ui/ducks/app/app.ts

```diff
@@ -1,10 +1,71 @@
+import { AnyAction } from 'redux';
 import {
   WebHIDConnectedStatuses,
   HardwareTransportStates,
 } from '../../../shared/constants/hardware-wallets';
 import * as actionConstants from '../../store/actionConstants';
 
-export default function reduceApp(state = {}, action) {
+interface ReducerState {
```

Comment:
@brad-decker wdyt of the name and approach here?

Reply:
I'll pull your branch tonight or tomorrow morning and give this a good review. looks promising from initial glance.